enhancements/external-control-plane-topology.md

```diff
@@ -0,0 +1,93 @@
+---
+title: External Control Plane Topology
+authors:
+  - "@csrwng"
+reviewers:
+  - "@derekwaynecarr"
+  - "@ironcladlou"
+  - "@enxebre"
+  - "@sjenning"
+  - "@sttts"
+  - "@deads2k"
+  - "@mfojtik"
+  - "@s-urbaniak"
+  - "@spadgett"
+  - "@dmage"
+  - "@Miciah"
+approvers:
+  - "@derekwaynecarr"
+  - "@smarterclayton"
+creation-date: 2021-04-19
+last-updated: 2021-04-19
+status: implementable
+see-also:
+  - "/enhancements/update/ibm-public-cloud-support.md"
+  - "/enhancements/single-node/cluster-high-availability-mode-api.md"
+---
+
+# External Control Plane Topology
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+External control plane support was introduced in OCP 4 with the support
+for the [IBM Cloud Managed service](https://github.com/openshift/enhancements/blob/master/enhancements/update/ibm-public-cloud-support.md) (ROKS). At the time, this was the only platform that
+was run with an external control plane. It was sufficient to use a platform type of
+`IBMCloud` to distinguish it from other OCP installations with traditional
+control plane topology.
+
+More recently, an orthogonal field was [added to the Infrastructure resource](https://github.com/openshift/enhancements/blob/master/enhancements/single-node/cluster-high-availability-mode-api.md) to indicate
+the type of control plane topology. Current supported values for the `controlPlaneTopology` field
+are `HighlyAvailable` and `SingleReplica`.
+
+This enhancement proposes adding a third possible value to the control plane topology
+field: `External`. A value of `External` in this field indicates that control plane components
+such as Etcd, Kube API server, Kube Controller Manager, and Kube Scheduler are running outside
+the cluster and are not visible as pods inside the cluster.
+
+## Motivation
+
+Whether the control plane is external or not should not be tied to the platform that the cluster
+is running on. IBM Cloud will soon support IPI/UPI installation. Thus, having a platform of `IBMCloud`
+will not imply an external control plane. Hypershift will bring support for external control planes
+to existing platforms such as AWS.
+
+### Goals
+
+- Allow expressing a new type of control plane topology in the Infrastructure resource inside an
+  OCP cluster.
+
+- Provide operators/components that change their behavior based on whether the control plane is external or
+  self hosted, a clear indicator of what mode they're running in.
+
+### Non-Goals
+
+- Provide a design for running OCP with an external control plane.
+
+- Describe how the `controlPlaneTopology` field will be set for an external control plane deployment.
```

Comment:
This enhancement is talking about the value it will be set to.  Are you saying that we will not specify the actor which sets the value?

Reply:
Yes, I didn't think it was relevant to explain that Hypershift or IBM ROKS toolkit do their own bootstrapping and will set the value appropriately. But if you think it's relevant I can add it to the doc.

Reply:
I'm fine leaving it out.  But maybe rephrase this line to:

> - Describe the mechanics of populating `controlPlaneTopology` the field.  This enhancement only introduces the new value.

or some such to make it more clear what portion is out of scope.

Reply:
@wking for clarification, the externalized control plane manager is responsible for projecting this value into the end-user cluster data plane.